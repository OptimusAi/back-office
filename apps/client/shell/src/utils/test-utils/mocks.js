// results.spec.js
export const licencePlatesMock = [
  {
    id: 1,
    number: 'TEST1',
    zones: [
      {
        id: '2003',
        infractions: [
          {
            id: 'testId01',
            observation: {
              id: 'testId',
              observedAt: new Date(),
              licencePlate: {
                licencePlateNumber: 'TEST1',
                countrySubdivision: {
                  name: 'ALBERTA',
                  code: 'AB',
                  countryCode: 'CA',
                  country: 'Canada',
                  regionType: 'PROVINCE',
                },
              },
              countrySubdivision: {
                name: 'ALBERTA',
                code: 'AB',
                countryCode: 'CA',
                country: 'Canada',
                regionType: 'PROVINCE',
              },
              zoneId: '001',
              photos: [
                { url: 'http://fakeurl.com', imageType: 'PROFILE' },
                { url: 'http://fakeurl.com', imageType: 'PLATE' },
              ],
            },
          },
        ],
        sessions: [{ id: 1, start: '', end: '' }],
      },
    ],
  },
  {
    id: 2,
    number: 'TEST2',
    zones: [
      {
        id: '2003',
        infractions: [
          {
            id: 'testId02',
            observation: {
              id: 'testId02',
              observedAt: new Date(),
              licencePlate: {
                licencePlateNumber: 'TEST1',
                countrySubdivision: {
                  name: 'ALBERTA',
                  code: 'AB',
                  countryCode: 'CA',
                  country: 'Canada',
                  regionType: 'PROVINCE',
                },
              },
              countrySubdivision: {
                name: 'ALBERTA',
                code: 'AB',
                countryCode: 'CA',
                country: 'Canada',
                regionType: 'PROVINCE',
              },
              zoneId: '002',
              photos: [
                { url: 'http://fakeurl.com', imageType: 'PROFILE' },
                { url: 'http://fakeurl.com', imageType: 'PLATE' },
              ],
            },
          },
        ],
        sessions: [{ id: 2, start: '', end: '' }],
      },
    ],
  },
];

// licence-plate.details.spec
export const activeInfractionOnEnforce = {
  id: 3,
  licencePlate: {
    licencePlateNumber: 'TESTPLT',
    countrySubdivision: {
      name: 'ALBERTA',
      code: 'AB',
      countryCode: 'CA',
      country: 'Canada',
      regionType: 'PROVINCE',
    },
  },
  observations: [
    {
      zone: 1,
      observedAt: new Date(),
      eventType: 'REALTIME',
    },
  ],
  zone: 1,
  address: 'Test Address',
  enforcementPhotos: [
    { url: 'http://fakeurl.com', imageType: 'PROFILE', useToEnforce: true },
    { url: 'http://fakeurl.com', imageType: 'PLATE', useToEnforce: true },
  ],
};

export const licencePlateImage = {
  photoId: '0b06b9c2-4e87-4bb6-adab-1972a732faeb',
  observationId: '55cc15ef-92df-4d2d-9e24-3002d24e0fc0',
  infractionId: '8e5aa863-a9e3-4b17-9b2e-59cbb1cde8eb',
  imageType: 'PROFILE',
  useToEnforce: false,
  url: 'https://parkplus/images/',
};

export const regions = [
  {
    code: 'QC',
    country: 'Canada',
    countryCode: 'CA',
    name: 'Quebec',
    regionType: 'PROVINCE',
  },
  {
    code: 'FL',
    country: 'United States',
    countryCode: 'US',
    name: 'Florida',
    regionType: 'STATE',
  },
  {
    code: 'WA',
    country: 'United States',
    countryCode: 'US',
    name: 'Washington',
    regionType: 'STATE',
  },
  {
    code: 'AB',
    country: 'Canada',
    countryCode: 'CA',
    name: 'Alberta',
    regionType: 'PROVINCE',
  },
];

// licence-plate.container.spec
export const activeInfractionMock = {
  testId01: {
    id: 'testId01',
    observation: {
      id: 'testId',
      observedAt: new Date(),
      licencePlate: {
        licencePlateNumber: 'TEST1',
        countrySubdivision: {
          name: 'ALBERTA',
          code: 'AB',
          countryCode: 'CA',
          country: 'Canada',
          regionType: 'PROVINCE',
        },
      },
      zoneId: '001',
      photos: [
        { url: 'http://fakeurl.com', imageType: 'PROFILE' },
        { url: 'http://fakeurl.com', imageType: 'PLATE' },
      ],
    },
    status: 'ENFORCEABLE',
    licencePlate: {
      licencePlateNumber: 'TEST1',
      countrySubdivision: {
        name: 'ALBERTA',
        code: 'AB',
        countryCode: 'CA',
        country: 'Canada',
        regionType: 'PROVINCE',
      },
    },
    enforcementPhotos: [
      {
        url: 'http://fakeurl.com',
        imageType: 'PROFILE',
        useToEnforce: true,
      },
      {
        url: 'http://fakeurl.com',
        imageType: 'PLATE',
        useToEnforce: true,
      },
    ],
  },
};

export const activeParkadeInfractionMock = {
  id: 'testId01',
  observations: [
    {
      id: '1',
      observedAt: new Date(),
      licencePlate: {
        licencePlateNumber: 'TEST1',
        countrySubdivision: {
          name: 'ALBERTA',
          code: 'AB',
          countryCode: 'CA',
          country: 'Canada',
          regionType: 'PROVINCE',
        },
      },
      eventType: 'EXIT',
      zone: '001',
      photos: [
        {
          photoId: '1',
          infractionId: 'testId01',
          observationId: '1',
          url: 'http://fakeurl.com',
          imageType: 'REAR',
          useToEnforce: true,
        },
        {
          photoId: '1a',
          infractionId: 'testId01',
          observationId: '1',
          url: 'http://fakeurl.com',
          imageType: 'PLATE',
          useToEnforce: true,
        },
      ],
    },
    {
      id: '2',
      observedAt: new Date(),
      licencePlate: {
        licencePlateNumber: 'TEST1',
        countrySubdivision: {
          name: 'ALBERTA',
          code: 'AB',
          countryCode: 'CA',
          country: 'Canada',
          regionType: 'PROVINCE',
        },
      },
      eventType: 'ENTRY',
      zone: '001',
      photos: [
        {
          photoId: '2',
          infractionId: 'testId01',
          observationId: '2',
          url: 'http://fakeurl.com',
          imageType: 'REAR',
          useToEnforce: true,
        },
        {
          photoId: '2a',
          infractionId: 'testId01',
          observationId: '1',
          url: 'http://fakeurl.com',
          imageType: 'PLATE',
          useToEnforce: true,
        },
      ],
    },
  ],
  status: 'ENFORCEABLE',
  licencePlate: {
    licencePlateNumber: 'TEST1',
    countrySubdivision: {
      name: 'ALBERTA',
      code: 'AB',
      countryCode: 'CA',
      country: 'Canada',
      regionType: 'PROVINCE',
    },
  },
  enforcementPhotos: [
    {
      photoId: '1',
      infractionId: 'testId01',
      observationId: '1',
      url: 'http://fakeurl.com',
      imageType: 'REAR',
      useToEnforce: true,
    },
    {
      photoId: '2',
      infractionId: 'testId01',
      observationId: '2',
      url: 'http://fakeurl.com',
      imageType: 'REAR',
      useToEnforce: true,
    },
    {
      photoId: '3',
      infractionId: 'testId01',
      observationId: '1',
      url: 'http://fakeurl.com',
      imageType: 'PLATE',
      useToEnforce: true,
    },
    {
      photoId: '4',
      infractionId: 'testId01',
      observationId: '2',
      url: 'http://fakeurl.com',
      imageType: 'PLATE',
      useToEnforce: true,
    },
  ],
  zone: '2003',
};

export const activeLicencePlateMock = {
  id: 1,
  number: 'TEST1',
  zones: [
    {
      id: '2003',
      infractions: [
        {
          id: 'testId01',
          observation: {
            id: 'testId',
            observedAt: new Date(),
            licencePlate: {
              licencePlateNumber: 'TEST1',
              countrySubdivision: {
                name: 'ALBERTA',
                code: 'AB',
                countryCode: 'CA',
                country: 'Canada',
                regionType: 'PROVINCE',
              },
            },
            countrySubdivision: {
              name: 'ALBERTA',
              code: 'AB',
              countryCode: 'CA',
              country: 'Canada',
              regionType: 'PROVINCE',
            },
            zoneId: '001',
            photos: [
              { url: 'http://fakeurl.com', imageType: 'PROFILE' },
              { url: 'http://fakeurl.com', imageType: 'PLATE' },
            ],
          },
        },
      ],
      sessions: [{ id: 1, start: '', end: '' }],
      permitSchedules: [
        { id: 1, effectiveDate: '', expiryDate: '', permit: { id: 1 } },
      ],
    },
  ],
};

// enforcement utils

export const permitMock = {
  id: '123abc',
  effectiveDate: '2022-01-01T07:00:00Z',
  expiryDate: '2023-01-01T07:00:00Z',
  licencePlate: 'ABC-123',
  zones: ['1001'],
  permit: {
    id: 'abc123',
    noOfStalls: 2,
    category: 'EXEMPT_VEHICLE',
    externalId: '123',
  },
  parkingSchedule: [
    { day: 'MONDAY', time: { start: '08:00:00', end: '18:00:00' } },
    { day: 'TUESDAY', time: { start: '08:00:00', end: '18:00:00' } },
    { day: 'WEDNESDAY', time: { start: '08:00:00', end: '18:00:00' } },
    { day: 'THURSDAY', time: { start: '08:00:00', end: '18:00:00' } },
    { day: 'FRIDAY', time: { start: '08:00:00', end: '18:00:00' } },
    { day: 'SATURDAY', time: { start: '00:00:00', end: '23:00:00' } },
    { day: 'SUNDAY', time: { start: '00:00:00', end: '23:00:00' } },
  ],
};

// ADMIN

export const loggedInUser = {
  firstName: 'John',
  lastName: 'Doe',
  email: 'jd@test.com',
  authorities: ['ROLE_ADMIN'],
};

export const roles = [
  {
    authorities: [{ authority: 'ROLE_ADMIN' }],
    externalId: '110022',
    externalIdIssuer: 'test',
    id: '306',
  },
];

export const users = [
  { emailAdress: 'test@mail.com', id: '102', parkPlusUserId: '306' },
];

export const mappedUsers = [
  {
    id: '306',
    authorities: ['ROLE_ADMIN'],
    emailAdress: 'test@mail.com',
    parkPlusUserId: '306',
    userId: '102',
  },
];

// ZONES

export const zonesMock = [
  {
    id: 1,
    zone: 1,
    latitude: 80.40821857,
    longitude: -110.87463,
    pairingTimeLimit: 15,
    arrivalGraceTime: 5,
    departureGraceTime: 10,
    zoneCategory: {
      id: 'z001',
      name: 'On Street',
      description: 'On street parking',
    },
    city: {
      id: '1',
      city: 'Calgary',
      countrySubdivision: {
        name: 'ALBERTA',
        code: 'AB',
        countryCode: 'CA',
        country: 'Canada',
        regionType: 'PROVINCE',
      },
    },
  },
];

export const formattedZonesMock = [
  {
    id: 1,
    zone: 1,
    latitude: 80.40821857,
    longitude: -110.87463,
    pairingTimeLimit: 15,
    arrivalGraceTime: 5,
    departureGraceTime: 10,
    zoneCategory: {
      id: 'z001',
      name: 'On Street',
      description: 'On street parking',
    },
    city: {
      id: '1',
      city: 'Calgary',
      countrySubdivision: {
        name: 'ALBERTA',
        code: 'AB',
        countryCode: 'CA',
        country: 'Canada',
        regionType: 'PROVINCE',
      },
    },
  },
];

export const zoneCategoriesMock = [
  {
    id: 'z001',
    name: 'On Street',
    description: 'On Street Parking',
  },
];

export const mappedZoneCategoriesMock = [
  {
    id: 'z001',
    zoneCategory: 'On Street',
    description: 'On Street Parking',
  },
];

export const zoneCitiesMock = [
  {
    id: '1',
    city: 'Calgary',
    countrySubdivision: {
      name: 'ALBERTA',
      code: 'AB',
      countryCode: 'CA',
      country: 'Canada',
      regionType: 'PROVINCE',
    },
  },
];

export const updatedZoneMock = {
  id: 1,
  zone: 1,
  latitude: 80.40821857,
  longitude: -110.87463,
  pairingTimeLimit: 10,
  arrivalGraceTime: 10,
  departureGraceTime: 10,
  zoneCategory: {
    id: 'z001',
    name: 'On Street',
    description: 'On street parking',
  },
  city: {
    id: '1',
    city: 'Calgary',
    countrySubdivision: {
      name: 'ALBERTA',
      code: 'AB',
      countryCode: 'CA',
      country: 'Canada',
      regionType: 'PROVINCE',
    },
  },
};

export const mappedZoneRowsMock = [
  {
    id: 1,
    zone: 1,
    zoneCity: 'Calgary',
    zoneCategory: 'On Street',
    pairingTimeLimit: 15,
    arrivalGraceTime: 5,
    departureGraceTime: 10,
  },
];

// BYLAWS

export const infractionBylaws = [
  {
    id: '2222',
    zoneCategoryId: '1dd2d96e',
    defaultBylaw: false,
    bylaw: {
      id: 222,
      code: 'test2',
      sectionCode: '9(21)',
      description: 'test2',
      externalReferenceId: 'eb222',
    },
  },
  {
    id: '1111',
    zoneCategoryId: '1dd2d96e',
    defaultBylaw: false,
    bylaw: {
      id: 111,
      code: 'test1',
      sectionCode: '1(13)',
      description: 'test1',
      externalReferenceId: 'eb111',
    },
  },
];

export const bylawsMock = [
  { id: 111, code: 'test1', sectionCode: 1, description: 'test1' },
  { id: 222, code: 'test2', sectionCode: 2, description: 'test2' },
];

export const mappedBylawsMock = [
  { id: 111, bylawCode: 'test1', sectionCode: 1, description: 'test1' },
  { id: 222, bylawCode: 'test2', sectionCode: 2, description: 'test2' },
];
