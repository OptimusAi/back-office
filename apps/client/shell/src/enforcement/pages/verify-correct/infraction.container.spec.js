import 'regenerator-runtime/runtime';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Infraction from './infraction.container';
import configureStore from 'redux-mock-store';
import SpecFriendly from '../../../spec-friendly';
import { zonesMock } from '../../../utils/test-utils/mocks';

const mockStore = configureStore([]);
describe('Infraction Container', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      app: {
        zones: {
          ids: [1],
          entities: { 1: { id: 1, ...zonesMock[0] } },
        },
        regions: {
          ids: [1],
          entities: { 1: { id: 1 } },
        },
      },
      permissions: {
        permissions: ['ROLE_ADMIN'],
      },
      enforcement: {
        infractions: {
          busy: false,
          error: null,
          ids: ['test1'],
          zones: {
            ids: [1],
            entities: { 1: zonesMock[0] },
          },
          zonesList: zonesMock,
          entities: {
            test1: {
              id: 'test1',
              observations: [
                {
                  id: 'test1',
                  observedAt: new Date(),
                  licencePlate: {
                    licencePlateNumber: 'TEST1',
                    countrySubdivision: {
                      name: 'ALBERTA',
                      code: 'AB',
                      countryCode: 'CA',
                      country: 'Canada',
                      regionType: 'PROVINCE',
                    },
                  },
                  zone: '001',
                  photos: [
                    { url: 'http://fakeurl.com', imageType: 'PROFILE' },
                    { url: 'http://fakeurl.com', imageType: 'PLATE' },
                  ],
                },
              ],
              status: 'ENFORCEABLE',
              zone: {
                zone: '1',
              },
              licencePlate: {
                licencePlateNumber: 'TEST1',
                countrySubdivision: {
                  name: 'ALBERTA',
                  code: 'AB',
                  countryCode: 'CA',
                  country: 'Canada',
                  regionType: 'PROVINCE',
                },
              },
              enforcementPhotos: [
                {
                  url: 'http://fakeurl.com',
                  imageType: 'PROFILE',
                  useToEnforce: true,
                },
                {
                  url: 'http://fakeurl.com',
                  imageType: 'PLATE',
                  useToEnforce: true,
                },
              ],
            },
          },
          activeInfractionId: 'test1',
          searchDate: null,
          activeImageViewer: false,
          activeImage: -1,
          originalActiveInfraction: null,
          activeInfraction: {
            id: 'test1',
            observations: [
              {
                id: 'test1',
                observedAt: new Date(),
                licencePlate: {
                  licencePlateNumber: 'TEST1',
                  countrySubdivision: {
                    name: 'ALBERTA',
                    code: 'AB',
                    countryCode: 'CA',
                    country: 'Canada',
                    regionType: 'PROVINCE',
                  },
                },
                zone: '001',
                photos: [
                  { url: 'http://fakeurl.com', imageType: 'PROFILE' },
                  { url: 'http://fakeurl.com', imageType: 'PLATE' },
                ],
              },
            ],
            status: 'ENFORCEABLE',
            zone: {
              zone: '1',
            },
            licencePlate: {
              licencePlateNumber: 'TEST1',
              countrySubdivision: {
                name: 'ALBERTA',
                code: 'AB',
                countryCode: 'CA',
                country: 'Canada',
                regionType: 'PROVINCE',
              },
            },
            enforcementPhotos: [
              {
                url: 'http://fakeurl.com',
                imageType: 'PROFILE',
                useToEnforce: true,
              },
              {
                url: 'http://fakeurl.com',
                imageType: 'PLATE',
                useToEnforce: true,
              },
            ],
          },
          isEditing: false,
        },
      },
    });

    component = renderer.create(
      <SpecFriendly>
        <Provider store={store}>
          <Infraction />
        </Provider>
      </SpecFriendly>
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
