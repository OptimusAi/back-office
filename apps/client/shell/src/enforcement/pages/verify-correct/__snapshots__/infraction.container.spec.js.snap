// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Infraction Container should render with given state from Redux store 1`] = `
<div
  className="makeStyles-container-1"
>
  <div
    className="makeStyles-container-2"
  >
    <div
      className="MuiToolbar-root makeStyles-toolbar-4 MuiToolbar-dense MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root makeStyles-title-5 MuiTypography-subtitle1 MuiTypography-noWrap"
      >
        Potential Infraction
      </h6>
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-helpIcon-6"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  <div
    className="makeStyles-container-7"
  >
    <div>
      <img
        alt=""
        className="makeStyles-licencePlateImage-8"
      />
    </div>
    <div
      className="makeStyles-firstColumn-10"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        onKeyDown={[Function]}
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink={true}
          htmlFor="prevent-outside-tab"
          id="prevent-outside-tab-label"
        >
          Licence Plate
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-describedby="prevent-outside-tab-helper-text"
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="prevent-outside-tab"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            style={
              Object {
                "fontSize": 30,
              }
            }
            type="text"
            value="TEST1"
          />
        </div>
        <p
          className="MuiFormHelperText-root MuiFormHelperText-filled"
          id="prevent-outside-tab-helper-text"
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </p>
      </div>
      <hr
        className="MuiDivider-root makeStyles-divider-14 MuiDivider-vertical"
      />
    </div>
    <div
      className="makeStyles-secondColumn-11"
    >
      <div
        className="MuiFormControl-root makeStyles-field-9 MuiFormControl-fullWidth"
      >
        <div
          aria-expanded={false}
          aria-owns={null}
          className="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon"
          margin="dense"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="combobox"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
              htmlFor="region"
              id="region-label"
            >
              Province/State
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-controls={null}
                aria-invalid={false}
                autoCapitalize="none"
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd"
                disabled={false}
                id="region"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                required={false}
                spellCheck="false"
                type="text"
                value=""
              />
              <div
                className="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Open"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Open"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root makeStyles-field-9 MuiFormControl-fullWidth"
      >
        <div
          aria-expanded={false}
          aria-owns={null}
          className="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasPopupIcon"
          margin="dense"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="combobox"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
              htmlFor="zone"
              id="zone-label"
            >
              Zone
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-controls={null}
                aria-invalid={false}
                autoCapitalize="none"
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd"
                disabled={false}
                id="zone"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                required={false}
                spellCheck="false"
                type="text"
                value=""
              />
              <div
                className="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Open"
                  className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Open"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root makeStyles-field-9 MuiFormControl-fullWidth"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          style={
            Object {
              "marginTop": "8px",
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
            data-shrink={true}
          >
            Address
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              readOnly={true}
              required={false}
              style={
                Object {
                  "cursor": "default",
                }
              }
              tabIndex={-1}
              type="text"
              value=" "
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="makeStyles-thirdColumn-12"
    >
      <div
        className="makeStyles-dinamicFields-13"
      />
      <div
        className="makeStyles-infractionActionsContainer-15"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-ignoreButton-16 MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeSmall"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root makeStyles-ignore-21"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            Ignore
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-enforceableButton-17 MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeSmall"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root makeStyles-enforceable-20"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
                />
              </svg>
            </span>
            Enforceable
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="makeStyles-imagesContainer-22"
  >
    <ul
      className="MuiImageList-root"
      style={
        Object {
          "margin": -2,
        }
      }
    >
      <li
        className="MuiImageListItem-root"
        style={
          Object {
            "height": "auto",
            "padding": 2,
            "width": "100%",
          }
        }
      >
        <div
          className="MuiImageListItem-item"
        >
          <div
            className="makeStyles-subtitle-26"
          >
            <h6
              className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
            >
              Available images
            </h6>
            <hr
              className="MuiDivider-root"
            />
          </div>
        </div>
      </li>
    </ul>
  </div>
  <span
    onFocus={[Function]}
    tabIndex="0"
  />
</div>
`;
